### PR TITLE
[TS] LPS-150021 |  View Usage for web contents show results from multiple scopes, where only the copy of the web content is used

### PR DESCRIPTION
**Pull request by bakayattila**
hi Team,
Could you review this?

https://issues.liferay.com/browse/LPS-150021

Best,
Attila

-----

**Reproduction steps:**

1. Set up a clean portal instance
2. Create a blank site named SiteA
3. Create a web content named ContentA
4. Create a widget page named PageA
5. Add a web content display portlet on PageA showing ContentA
6. Export the entire SiteA site, download the LAR file
7. Create a new site names SiteB
8. Import the LAR file in SiteB
9. Go to the web content list in SiteB and select "View Usages" of ContentA

 **Expected result:** Only one usage is shown: the one in SiteB
 **Actual result:** two usages are shown: one in SiteA and one in SiteB (you can check the links in the "View in Page" option on the usage entry)

The modification was required to avoid conflict of usage count between the same article in different scopes. If an article is imported to a site in the same portal the article will keep the ArticleId but it will be a different article from the original.

After the modification, I re-tested by importing the export to a new site, only the relevant usages were shown. I also tested if the search could find usages in different scopes by creating an article in the global scope and using it in both site A and site B. Even after my modification, the usages were shown correctly for the article in the global scope.

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by bakayattila** [comment]
ci:test:sf

File: modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/asset/util/JournalArticleLayoutClassedModelUsageRecorder.java [hunk 1]
```diff
@@ -105,7 +105,7 @@ private void _recordJournalContentSearches(
 
 		List<JournalContentSearch> contentSearches =
 			_journalContentSearchLocalService.getArticleContentSearches(
-				article.getArticleId());
+				article.getGroupId(), article.getArticleId());
```
Comment (lfbesada) [inline modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/asset/util/JournalArticleLayoutClassedModelUsageRecorder.java:108]: Hi @bakayattila,
I think that by introducing this change maybe we would be stopping to take into account the use of journal articles in layouts that do not belong to the site. For example, any use of a global site journal article, or the use of a shared journal article in an asset library on another site. What do you think?
Thanks

**Review by lfbesada — CHANGES_REQUESTED** [review]

File: modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/asset/util/JournalArticleLayoutClassedModelUsageRecorder.java (hunk 1 above)
Comment (bakayattila) [reply modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/asset/util/JournalArticleLayoutClassedModelUsageRecorder.java:108]: Hi @lfbesada,
I had that fear as well. That is why I tested it with the global scope as I stated in the first comment:

> After the modification, I re-tested by importing the export to a new site, only the relevant usages were shown. I also tested if the search could find usages in different scopes by creating an article in the global scope and using it in both site A and site B. Even after my modification, the usages were shown correctly for the article in the global scope.

Comment (lfbesada) [reply modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/asset/util/JournalArticleLayoutClassedModelUsageRecorder.java:108]: @bakayattila thanks for the explanation. Did you test the case of content shared between asset libraries?

Comment (bakayattila) [reply modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/asset/util/JournalArticleLayoutClassedModelUsageRecorder.java:108]: No, I only tested the global site and the specific use case.

**Review by lfbesada — APPROVED** [review]
Hi @bakayattila 
Thank you very much for the fix and special thanks for such a simple and elegant solution. :clap: According to my tests, it works great!
I was amazed that you found such a simple solution. I had analyzed this bug before and the solutions I had found were quite complex.
Thank you very much!

**Comment by lfbesada** [comment]
ci:forward

**Pull request closed without merging**